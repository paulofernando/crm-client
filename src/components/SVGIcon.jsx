import React from "react";

const getPath = (name, props) => {
  switch (name) {
    case "logo":
        return (
          <path
            {...props}
            d="M0 22h12v2h-12v-2zm11-1h-10c0-1.105.895-2 2-2h6c1.105 0 2 .895 2 2zm6.369-12.839l-2.246 2.197s6.291 5.541 8.172 7.144c.475.405.705.929.705 1.446 0 1.015-.888 1.886-1.95 1.819-.52-.032-.981-.303-1.321-.697-1.619-1.875-7.07-8.249-7.07-8.249l-2.245 2.196-5.857-5.856 5.957-5.857 5.855 5.857zm-12.299.926c-.195-.193-.458-.302-.733-.302-.274 0-.537.109-.732.302-.193.195-.303.458-.303.733 0 .274.11.537.303.732l5.513 5.511c.194.195.457.304.732.304.275 0 .538-.109.732-.304.194-.193.303-.457.303-.732 0-.274-.109-.537-.303-.731l-5.512-5.513zm8.784-8.784c-.195-.194-.458-.303-.732-.303-.576 0-1.035.467-1.035 1.035 0 .275.108.539.303.732l5.513 5.513c.194.193.456.303.731.303.572 0 1.036-.464 1.036-1.035 0-.275-.109-.539-.304-.732l-5.512-5.513z"
          />
        );
    case "list":
      return (
        <path
          {...props}
          d="M24 6h-24v-4h24v4zm0 4h-24v4h24v-4zm0 8h-24v4h24v-4z"
        />
      );
    case "create":
      return (
        <path
          {...props}
          d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm7 14h-5v5h-4v-5h-5v-4h5v-5h4v5h5v4z"
        />
      );
    case "search":
      return (
        <path
          {...props}
          d="M23.111 20.058l-4.977-4.977c.965-1.52 1.523-3.322 1.523-5.251 0-5.42-4.409-9.83-9.829-9.83-5.42 0-9.828 4.41-9.828 9.83s4.408 9.83 9.829 9.83c1.834 0 3.552-.505 5.022-1.383l5.021 5.021c2.144 2.141 5.384-1.096 3.239-3.24zm-20.064-10.228c0-3.739 3.043-6.782 6.782-6.782s6.782 3.042 6.782 6.782-3.043 6.782-6.782 6.782-6.782-3.043-6.782-6.782zm2.01-1.764c1.984-4.599 8.664-4.066 9.922.749-2.534-2.974-6.993-3.294-9.922-.749z"
        />
      );
    case "grid":
      return (
        <path
          {...props}
          d="M6 6h-6v-6h6v6zm9-6h-6v6h6v-6zm9 0h-6v6h6v-6zm-18 9h-6v6h6v-6zm9 0h-6v6h6v-6zm9 0h-6v6h6v-6zm-18 9h-6v6h6v-6zm9 0h-6v6h6v-6zm9 0h-6v6h6v-6z"
        />
      );      
    case "loading":
      return (
        <path
          {...props}
          d="M13.75 22c0 .966-.783 1.75-1.75 1.75s-1.75-.784-1.75-1.75.783-1.75 1.75-1.75 1.75.784 1.75 1.75zm-1.75-22c-1.104 0-2 .896-2 2s.896 2 2 2 2-.896 2-2-.896-2-2-2zm10 10.75c.689 0 1.249.561 1.249 1.25 0 .69-.56 1.25-1.249 1.25-.69 0-1.249-.559-1.249-1.25 0-.689.559-1.25 1.249-1.25zm-22 1.25c0 1.105.896 2 2 2s2-.895 2-2c0-1.104-.896-2-2-2s-2 .896-2 2zm19-8c.551 0 1 .449 1 1 0 .553-.449 1.002-1 1-.551 0-1-.447-1-.998 0-.553.449-1.002 1-1.002zm0 13.5c.828 0 1.5.672 1.5 1.5s-.672 1.501-1.502 1.5c-.826 0-1.498-.671-1.498-1.499 0-.829.672-1.501 1.5-1.501zm-14-14.5c1.104 0 2 .896 2 2s-.896 2-2.001 2c-1.103 0-1.999-.895-1.999-2s.896-2 2-2zm0 14c1.104 0 2 .896 2 2s-.896 2-2.001 2c-1.103 0-1.999-.895-1.999-2s.896-2 2-2z"
        />
      );
    case "linked":
      return (
        <path
          {...props}
          d="M6.188 8.719c.439-.439.926-.801 1.444-1.087 2.887-1.591 6.589-.745 8.445 2.069l-2.246 2.245c-.644-1.469-2.243-2.305-3.834-1.949-.599.134-1.168.433-1.633.898l-4.304 4.306c-1.307 1.307-1.307 3.433 0 4.74 1.307 1.307 3.433 1.307 4.74 0l1.327-1.327c1.207.479 2.501.67 3.779.575l-2.929 2.929c-2.511 2.511-6.582 2.511-9.093 0s-2.511-6.582 0-9.093l4.304-4.306zm6.836-6.836l-2.929 2.929c1.277-.096 2.572.096 3.779.574l1.326-1.326c1.307-1.307 3.433-1.307 4.74 0 1.307 1.307 1.307 3.433 0 4.74l-4.305 4.305c-1.311 1.311-3.44 1.3-4.74 0-.303-.303-.564-.68-.727-1.051l-2.246 2.245c.236.358.481.667.796.982.812.812 1.846 1.417 3.036 1.704 1.542.371 3.194.166 4.613-.617.518-.286 1.005-.648 1.444-1.087l4.304-4.305c2.512-2.511 2.512-6.582.001-9.093-2.511-2.51-6.581-2.51-9.092 0z"
        />
      );      
    case "unlinked":
      return (
        <path
          {...props}
          d="M14.008 21.092l-3.392 1.959c-3.384 1.954-7.713.794-9.668-2.592-1.953-3.385-.793-7.713 2.591-9.667l3.393-1.959 1.495 2.589-3.393 1.96c-1.95 1.125-2.622 3.631-1.495 5.581 1.126 1.953 3.631 2.625 5.582 1.495l3.392-1.956 1.495 2.59zm-3.214-17.553l-1.959 3.392 2.589 1.497 1.959-3.393c1.126-1.95 3.631-2.622 5.581-1.495 1.952 1.127 2.624 3.629 1.499 5.582l-1.96 3.392 2.589 1.494 1.96-3.391c1.952-3.387.792-7.714-2.59-9.667-3.387-1.955-7.716-.795-9.668 2.589zm7.398 20.217l-1.724.244-.523-3.688 1.722-.243.525 3.687zm5.808-7.284l-.249 1.724-3.688-.527.247-1.722 3.69.525zm-5.733 3.013l1.229-1.23 2.638 2.636-1.231 1.23-2.636-2.636z"
        />
      );
    default:
      return <path />;
  }
};

const SVGIcon = ({
  name = "",
  fill = "#eee",
  width = "100%",
  height = "100%"
}) => (
  <svg
    width={width}
    height={height}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 24 24"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    {getPath(name, { fill })}
  </svg>
);

export default SVGIcon;
