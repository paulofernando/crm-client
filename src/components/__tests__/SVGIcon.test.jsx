import React from 'react';
import { shallow } from 'enzyme';
import SVGIcon from '../SVGIcon';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('SVGIcon', () => {

    it('renders correctly', () => {
        let wrapper = shallow(<SVGIcon name={"logo"} width="24" />);
        expect(wrapper).toMatchSnapshot();

        wrapper = shallow(<SVGIcon name={"logo"} width="24" fill={"#aaa"}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
});