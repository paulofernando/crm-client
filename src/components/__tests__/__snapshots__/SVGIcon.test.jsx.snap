// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGIcon renders correctly 1`] = `ShallowWrapper {}`;

exports[`SVGIcon renders correctly 2`] = `ShallowWrapper {}`;
